app/models/physical_storage.rb

```diff
@@ -39,4 +44,81 @@ def refresh_ems
 
     EmsRefresh.queue_refresh(ext_management_system)
   end
+
+  def self.create_physical_storage_queue(userid, ext_management_system, options = {})
+    task_opts = {
+      :action => "creating Cloud PhysicalStorage for user #{userid}",
+      :userid => userid
+    }
+
+    queue_opts = {
+      :class_name  => 'PhysicalStorage',
+      :method_name => 'create_physical_storage',
+      :role        => 'ems_operations',
```

Comment:
Is this actually an ems_operation?  It is really just creating a database record, or is it actually hitting the provider to perform some operation?  If not hitting the provider then in that case why do we need to queue this if we're just creating/editing a new record?

I think you'll probably need some credential validation method that has to return successful prior to the form being able to be submitted

Reply:
This is a provider operation. The raw_create goes to the provider and asks it to attach and start managing the new storage-system. 
The provider then logs into the system and gathers initial information about it, such as what pools it's already got defined. 
On the next inventory sync, the storage system and all its related data that's been collected by the provider is put into the MIQ DB.

It's not trivial to check credentials for a storage system because they don't all have a uniform login method. We'd need to add a new endpoint on our provider just for validation. 
If the credentials are wrong the provider won't be able to attach. The MIQ worker will get error 400 and fail in its task, and the system won't appear in inventory scanning.

So maybe we can skip the credential validation for this version?

Reply:
> This is a provider operation. The raw_create goes to the provider and asks it to attach and start managing the new storage-system.

:+1: 

> It's not trivial to check credentials for a storage system because they don't all have a uniform login method

This should be abstracted by the provider so it is consistent to the MIQ plugin, but if you want to push that off that's fine